### PR TITLE
Utilizes the sql server name

### PR DESCRIPTION
**Pull request by qwertyquerty**
So for this to work @crossedfall, you're gonna have to update our current database schema then add the identifier of our old server to all of the rows.

Untested

### CODE REVIEW

File: SQL/beestation_schema.sql [hunk 1]
```diff
@@ -47,6 +47,7 @@ DROP TABLE IF EXISTS `SS13_ban`;
 CREATE TABLE IF NOT EXISTS `SS13_ban` (
   `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
   `bantime` datetime NOT NULL,
+  `server` varchar(32) DEFAULT NULL,
```
Comment (Crossedfall) [inline SQL/beestation_schema.sql]:
```suggestion
  `server_name` varchar(32) DEFAULT NULL,
```

Should use `server_name` to keep in line with the standard naming convention.

File: SQL/beestation_schema.sql [hunk 2]
```diff
@@ -117,6 +118,7 @@ DROP TABLE IF EXISTS `SS13_connection_log`;
 CREATE TABLE IF NOT EXISTS `SS13_connection_log` (
   `id` int(11) NOT NULL AUTO_INCREMENT,
   `datetime` datetime DEFAULT NULL,
+  `server` varchar(32) DEFAULT NULL,
```
Comment (Crossedfall) [inline SQL/beestation_schema.sql]:
```suggestion
  `server_name` varchar(32) DEFAULT NULL,
```

File: SQL/beestation_schema.sql [hunk 3]
```diff
@@ -137,6 +139,7 @@ CREATE TABLE IF NOT EXISTS `SS13_death` (
   `y_coord` smallint(5) unsigned NOT NULL,
   `z_coord` smallint(5) unsigned NOT NULL,
   `mapname` varchar(32) NOT NULL,
+  `server` varchar(32) DEFAULT NULL,
```
Comment (Crossedfall) [inline SQL/beestation_schema.sql]:
```suggestion
  `server_name` varchar(32) DEFAULT NULL,
```

File: SQL/beestation_schema.sql [hunk 4]
```diff
@@ -195,6 +198,7 @@ CREATE TABLE IF NOT EXISTS `SS13_legacy_population` (
   `playercount` int(11) DEFAULT NULL,
   `admincount` int(11) DEFAULT NULL,
   `time` datetime NOT NULL,
+  `server` varchar(32) DEFAULT NULL,
```
Comment (Crossedfall) [inline SQL/beestation_schema.sql]:
```suggestion
  `server_name` varchar(32) DEFAULT NULL,
```

File: SQL/beestation_schema.sql [hunk 5]
```diff
@@ -442,6 +446,7 @@ CREATE TABLE IF NOT EXISTS `SS13_round` (
   `start_datetime` datetime DEFAULT NULL,
   `shutdown_datetime` datetime DEFAULT NULL,
   `end_datetime` datetime DEFAULT NULL,
+  `server` varchar(32) DEFAULT NULL,
```
Comment (Crossedfall) [inline SQL/beestation_schema.sql]:
```suggestion
  `server_name` varchar(32) DEFAULT NULL,
```

File: code/controllers/subsystem/blackbox.dm [hunk 6]
```diff
@@ -46,7 +46,10 @@ SUBSYSTEM_DEF(blackbox)
 		if(M.client)
 			playercount += 1
 	var/admincount = GLOB.admins.len
-	var/datum/DBQuery/query_record_playercount = SSdbcore.NewQuery("INSERT INTO [format_table_name("legacy_population")] (playercount, admincount, time, server_ip, server_port, round_id) VALUES ([playercount], [admincount], '[SQLtime()]', INET_ATON(IF('[world.internet_address]' LIKE '', '0', '[world.internet_address]')), '[world.port]', '[GLOB.round_id]')")
+
+	var/ssqlname = sanitizeSQL(CONFIG_GET(string/serversqlname))
+
+	var/datum/DBQuery/query_record_playercount = SSdbcore.NewQuery("INSERT INTO [format_table_name("legacy_population")] (playercount, admincount, time, server, server_ip, server_port, round_id) VALUES ([playercount], [admincount], '[SQLtime()]', '[ssqlname]', INET_ATON(IF('[world.internet_address]' LIKE '', '0', '[world.internet_address]')), '[world.port]', '[GLOB.round_id]')")
```
Comment (Crossedfall) [inline code/controllers/subsystem/blackbox.dm]:
```suggestion
	var/datum/DBQuery/query_record_playercount = SSdbcore.NewQuery("INSERT INTO [format_table_name("legacy_population")] (playercount, admincount, time, server_name, server_ip, server_port, round_id) VALUES ([playercount], [admincount], '[SQLtime()]', '[ssqlname]', INET_ATON(IF('[world.internet_address]' LIKE '', '0', '[world.internet_address]')), '[world.port]', '[GLOB.round_id]')")
```

Needs to be updated to reflect the above

File: code/controllers/subsystem/blackbox.dm [hunk 7]
```diff
@@ -331,7 +335,9 @@ Versioning
...
+	ssqlname = sanitizeSQL(ssqlname)
+
+	var/datum/DBQuery/query_report_death = SSdbcore.NewQuery("INSERT INTO [format_table_name("death")] (pod, x_coord, y_coord, z_coord, mapname, server, server_ip, server_port, round_id, tod, job, special, name, byondkey, laname, lakey, bruteloss, fireloss, brainloss, oxyloss, toxloss, cloneloss, staminaloss, last_words, suicide) VALUES ('[sqlpod]', '[x_coord]', '[y_coord]', '[z_coord]', '[map]', '[ssqlname]', INET_ATON(IF('[world.internet_address]' LIKE '', '0', '[world.internet_address]')), '[world.port]', [GLOB.round_id], '[SQLtime()]', '[sqljob]', '[sqlspecial]', '[sqlname]', '[sqlkey]', '[laname]', '[lakey]', [sqlbrute], [sqlfire], [sqlbrain], [sqloxy], [sqltox], [sqlclone], [sqlstamina], '[last_words]', [suicide])")
```
Comment (Crossedfall) [inline code/controllers/subsystem/blackbox.dm]:
```suggestion
	var/datum/DBQuery/query_report_death = SSdbcore.NewQuery("INSERT INTO [format_table_name("death")] (pod, x_coord, y_coord, z_coord, mapname, server_name, server_ip, server_port, round_id, tod, job, special, name, byondkey, laname, lakey, bruteloss, fireloss, brainloss, oxyloss, toxloss, cloneloss, staminaloss, last_words, suicide) VALUES ('[sqlpod]', '[x_coord]', '[y_coord]', '[z_coord]', '[map]', '[ssqlname]', INET_ATON(IF('[world.internet_address]' LIKE '', '0', '[world.internet_address]')), '[world.port]', [GLOB.round_id], '[SQLtime()]', '[sqljob]', '[sqlspecial]', '[sqlname]', '[sqlkey]', '[laname]', '[lakey]', [sqlbrute], [sqlfire], [sqlbrain], [sqloxy], [sqltox], [sqlclone], [sqlstamina], '[last_words]', [suicide])")
```

File: code/controllers/subsystem/dbcore.dm [hunk 8]
```diff
@@ -120,7 +120,10 @@ SUBSYSTEM_DEF(dbcore)
 /datum/controller/subsystem/dbcore/proc/SetRoundID()
 	if(!Connect())
 		return
-	var/datum/DBQuery/query_round_initialize = SSdbcore.NewQuery("INSERT INTO [format_table_name("round")] (initialize_datetime, server_ip, server_port) VALUES (Now(), INET_ATON(IF('[world.internet_address]' LIKE '', '0', '[world.internet_address]')), '[world.port]')")
+
+	var/ssqlname = sanitizeSQL(CONFIG_GET(string/serversqlname))
+
+	var/datum/DBQuery/query_round_initialize = SSdbcore.NewQuery("INSERT INTO [format_table_name("round")] (initialize_datetime, server, server_ip, server_port) VALUES (Now(), '[ssqlname]', INET_ATON(IF('[world.internet_address]' LIKE '', '0', '[world.internet_address]')), '[world.port]')")
```
Comment (Crossedfall) [inline code/controllers/subsystem/dbcore.dm]:
```suggestion
	var/datum/DBQuery/query_round_initialize = SSdbcore.NewQuery("INSERT INTO [format_table_name("round")] (initialize_datetime, server_name, server_ip, server_port) VALUES (Now(), '[ssqlname]', INET_ATON(IF('[world.internet_address]' LIKE '', '0', '[world.internet_address]')), '[world.port]')")
```

File: code/modules/admin/sql_ban_system.dm [hunk 9]
```diff
@@ -463,9 +463,13 @@
 	var/adminwho = admins_online.Join(", ")
 	var/kn = key_name(usr)
 	var/kna = key_name_admin(usr)
+
+	var/ssqlname = sanitizeSQL(CONFIG_GET(string/serversqlname))
+
 	var/sql_ban
 	for(var/role in roles_to_ban)
 		sql_ban += list(list("bantime" = "NOW()",
+		"server" = "'[ssqlname]'",
```
Comment (Crossedfall) [inline code/modules/admin/sql_ban_system.dm]:
```suggestion
		"server_name" = "'[ssqlname]'",
```

File: code/modules/client/client_procs.dm [hunk 10]
```diff
@@ -581,7 +581,10 @@ GLOBAL_LIST_EMPTY(external_rsc_urls)
 		qdel(query_log_player)
 	if(!account_join_date)
 		account_join_date = "Error"
-	var/datum/DBQuery/query_log_connection = SSdbcore.NewQuery("INSERT INTO `[format_table_name("connection_log")]` (`id`,`datetime`,`server_ip`,`server_port`,`round_id`,`ckey`,`ip`,`computerid`) VALUES(null,Now(),INET_ATON(IF('[world.internet_address]' LIKE '', '0', '[world.internet_address]')),'[world.port]','[GLOB.round_id]','[sql_ckey]',INET_ATON('[sql_ip]'),'[sql_computerid]')")
+
+	var/ssqlname = sanitizeSQL(CONFIG_GET(string/serversqlname))
+
+	var/datum/DBQuery/query_log_connection = SSdbcore.NewQuery("INSERT INTO `[format_table_name("connection_log")]` (`id`,`datetime`,`server`,`server_ip`,`server_port`,`round_id`,`ckey`,`ip`,`computerid`) VALUES(null,Now(),'[ssqlname]',INET_ATON(IF('[world.internet_address]' LIKE '', '0', '[world.internet_address]')),'[world.port]','[GLOB.round_id]','[sql_ckey]',INET_ATON('[sql_ip]'),'[sql_computerid]')")
```
Comment (Crossedfall) [inline code/modules/client/client_procs.dm]:
```suggestion
	var/datum/DBQuery/query_log_connection = SSdbcore.NewQuery("INSERT INTO `[format_table_name("connection_log")]` (`id`,`datetime`,`server_name`,`server_ip`,`server_port`,`round_id`,`ckey`,`ip`,`computerid`) VALUES(null,Now(),'[ssqlname]',INET_ATON(IF('[world.internet_address]' LIKE '', '0', '[world.internet_address]')),'[world.port]','[GLOB.round_id]','[sql_ckey]',INET_ATON('[sql_ip]'),'[sql_computerid]')")
```

**Review by Crossedfall — CHANGES_REQUESTED** [review]
Update the column name to `server_name` to match the current naming convention and I think we'll be good. I'll apply it to Sage first.

**Comment by qwertyquerty** [comment]
Worth noting that the messages table already used the sql server name thing, and it had it under the field `server`

**Comment by Crossedfall** [comment]
> 
> 
> Worth noting that the messages table already used the sql server name thing, and it had it under the field `server`

hnng why must they break their own naming convention?

**Review by Crossedfall — APPROVED** [review]

**Comment by qwertyquerty** [comment]
Are you ready for this to be merged @crossedfall?

**Comment by Crossedfall** [comment]
Fully applied on Sage and Golden. No runtimes and everything looks to be working as expected.

**Pull request merged**